core/vmutils/src/main/java/org/arakhne/afc/vmutil/FileSystem.java

```diff
@@ -978,10 +976,8 @@ public static URL join(URL urlBase, File... elements) {
 		}
 		final StringBuilder buf = new StringBuilder(urlBase.getPath());
 		for (final File elt : elements) {
-			if (!elt.isAbsolute()) {
-				if (buf.length() == 0 || buf.charAt(buf.length() - 1) != URL_PATH_SEPARATOR_CHAR) {
-					buf.append(URL_PATH_SEPARATOR_CHAR);
-				}
+			if (!elt.isAbsolute() && buf.length() == 0 || buf.charAt(buf.length() - 1) != URL_PATH_SEPARATOR_CHAR) {
```

Comment:
This line should be:
```java
if (!elt.isAbsolute() && (buf.length() == 0 || buf.charAt(buf.length() - 1) != URL_PATH_SEPARATOR_CHAR)) {
```

Otherwise the behavior of the function is not the same as before the change.